File: artifacts/uniops/src/types/pod.ts
export type PodPhase = 'Pending' | 'Running' | 'Succeeded' | 'Failed' | 'Unknown' | 'Terminating';

export type PodQosClass = 'Guaranteed' | 'Burstable' | 'BestEffort';

export interface ContainerStatus {
  name: string;
  ready: boolean;
  restartCount: number;
  state: 'running' | 'waiting' | 'terminated';
  image: string;
}

export interface ResourceMetrics {
  cpuUsage: number;
  cpuLimit: number;
  cpuRequest: number;
  memoryUsage: number;
  memoryLimit: number;
  memoryRequest: number;
}

export interface Pod {
  id: string;
  name: string;
  namespace: string;
  phase: PodPhase;
  ready: boolean;
  readyContainers: number;
  totalContainers: number;
  containers: ContainerStatus[];
  nodeName: string;
  podIP: string;
  qosClass: PodQosClass;
  resources: ResourceMetrics;
  labels: Record<string, string>;
  annotations?: Record<string, string>;
  createdAt: string;
  restartCount: number;
  age: string;
  clusterName: string;
  companyId: string;
  integrationId: string;
}

export interface ClusterNode {
  name: string;
  status: 'Ready' | 'NotReady' | 'Unknown';
  roles: string[];
  version: string;
  cpu: number;
  memory: number;
  pods: number;
  maxPods: number;
}

export interface ClusterSummary {
  name: string;
  version: string;
  nodes: ClusterNode[];
  totalPods: number;
  runningPods: number;
  failedPods: number;
  namespaces: string[];
}

File: backend/app/models/pod.py
from sqlalchemy import BigInteger, Float, Integer, String
from sqlalchemy.orm import Mapped, mapped_column
from app.models.base import BaseModel


class Pod(BaseModel):
    __tablename__ = "pods"

    tenant_id: Mapped[str] = mapped_column(String(36), ForeignKey("tenants.id"), nullable=False)
    integration_id: Mapped[str | None] = mapped_column(String(36), ForeignKey("integrations.id"))
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    namespace: Mapped[str] = mapped_column(String(255), default="default")
    cluster: Mapped[str] = mapped_column(String(255))
    status: Mapped[str] = mapped_column(String(50))
    phase: Mapped[str | None] = mapped_column(String(50))
    node: Mapped[str | None] = mapped_column(String(255))
    cpu_request: Mapped[float | None] = mapped_column(Float)
    cpu_limit: Mapped[float | None] = mapped_column(Float)
    cpu_usage: Mapped[float | None] = mapped_column(Float)
    memory_request: Mapped[int | None] = mapped_column(BigInteger)
    memory_limit: Mapped[int | None] = mapped_column(BigInteger)
    memory_usage: Mapped[int | None] = mapped_column(BigInteger)
    restart_count: Mapped[int] = mapped_column(Integer, default=0)
    containers: Mapped[list] = mapped_column(JSON, default=list)
    labels: Mapped[dict] = mapped_column(JSON, default=dict)
